File: articles/coastal-flood-risk-baseline-assumptions/c/exposure_threshold.c
#include <stdio.h>
#include <stdbool.h>

bool is_exposed(
    double modeled_baseline,
    double baseline_correction,
    double sea_level_rise,
    double tide_surge,
    double uncertainty_margin,
    double land_elevation,
    double protection_height
) {
    double water_height = modeled_baseline + baseline_correction + sea_level_rise + tide_surge + uncertainty_margin;
    double threshold = land_elevation + protection_height;
    return water_height >= threshold;
}

int main(void) {
    bool exposed = is_exposed(0.0, 0.30, 0.45, 0.55, 0.10, 1.25, 0.20);
    printf("Synthetic site exposed after baseline correction: %s\n", exposed ? "true" : "false");
    return 0;
}

File: articles/coastal-flood-risk-baseline-assumptions/fortran/exposure_threshold.f90
program exposure_threshold
  implicit none

  real :: modeled_baseline, baseline_correction, sea_level_rise
  real :: tide_surge, uncertainty_margin, land_elevation, protection_height
  real :: water_height, threshold
  logical :: exposed

  modeled_baseline = 0.0
  baseline_correction = 0.30
  sea_level_rise = 0.45
  tide_surge = 0.55
  uncertainty_margin = 0.10
  land_elevation = 1.25
  protection_height = 0.20

  water_height = modeled_baseline + baseline_correction + sea_level_rise + tide_surge + uncertainty_margin
  threshold = land_elevation + protection_height
  exposed = water_height >= threshold

  print *, "Synthetic site exposed after baseline correction:", exposed
end program exposure_threshold
